app/src/main/java/com/cattailsw/mediaplayer/MainActivity.kt

```diff
@@ -53,12 +48,6 @@ class MainActivity : ComponentActivity() {
         lifecycleScope.launch {
             viewModel.state.collectLatest {
                 when (it) {
-                    is MainState.OpenFile -> {
-                        with(Dispatchers.Main) {
-                            startActivityForResult(it.intentToLaunch, 42)
-                        }
-                    }
-
                     MainState.ErrorOpen -> {
                         Toast.makeText(this@MainActivity, "open failed", Toast.LENGTH_SHORT).show()
```

Comment:
let's figure out how to do this in the composable so we can remove this state collection here